packages/node-sdk/src/client.ts

```diff
@@ -0,0 +1,607 @@
+      context: opts?.meta,
+    });
+  }
+
+  /**
+   * Tracks an event in Bucket.
+   *
+   * @param event - The event to track.
+   * @param opts.attributes - The attributes of the event (optional).
+   * @param opts.meta - The meta context associated with tracking (optional).
+   *
+   * @returns A boolean indicating if the request was successful.
+   * @throws An error if the event is invalid or the options are invalid.
+   * @remarks
+   * If the user is set, the event will be associated with the user.
+   * If the company is set, the event will be associated with the company.
+   **/
+  public async trackFeatureUsage(event: string, opts?: TrackOptions) {
+    ok(typeof event === "string" && event.length > 0, "event must be a string");
+
+    ok(opts === undefined || isObject(opts), "opts must be an object");
+    ok(
+      opts?.attributes === undefined || isObject(opts.attributes),
+      "attributes must be an object",
+    );
+    ok(
+      opts?.meta === undefined || isObject(opts.meta),
+      "meta must be an object",
+    );
+
+    return await this.post("event", {
+      event,
+      companyId: this._company?.companyId,
+      userId: this._user?.userId,
+      attributes: opts?.attributes,
+      context: opts?.meta,
+    });
+  }
+
+  /**
+   * Initializes the client by caching the feature flag definitions.
+   *
+   * @returns void
+   *
+   * @remarks
+   * Call this method before calling `getFlags` to ensure the feature flag definitions are cached.
+   **/
+  public async initialize() {
+    await this.getFeatureFlagDefinitionCache().refresh();
+  }
+
+  /**
+   * Gets the evaluated feature flags for the current context which includes the user, company, and custom context.
+   *
+   * @typeparam TFlagKey - The type of the feature flag keys, `string` by default.
+   *
+   * @returns The evaluated feature flags.
+   * @remarks
+   * Call `initialize` before calling this method to ensure the feature flag definitions are cached, empty flags will be returned otherwise.
+   **/
+  public getFlags<TFlagKey extends string = string>(): Readonly<
+    Record<TFlagKey, Flag>
+  > {
+    const mergedContext = {
+      user: this._user && {
+        id: this._user.userId,
+        ...this._user.attrs,
+      },
+      company: this._company && {
+        id: this._company.companyId,
+        ...this._company.attrs,
+      },
+      ...this._otherContext,
+    };
+
+    const flagDefinitions = this.getFeatureFlagDefinitionCache().get();
+    let evaluatedFlags: Flags<TFlagKey> =
+      (this._shared.fallbackFlags as Flags<TFlagKey>) || {};
+
+    if (flagDefinitions) {
+      const keyToVersionMap = new Map<string, number>(
+        flagDefinitions.flags.map((f) => [f.key, f.version]),
+      );
+
+      const evaluated = flagDefinitions.flags.map((flag) =>
+        evaluateFlag({ context: mergedContext, flag }),
+      );
+
+      evaluated.forEach(async (res) => {
+        await this.sendFeatureFlagEvent({
```

Comment:
this should probably also be rateLimited in the same way that "check" events are?